test/logic_name_test.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:rohd/rohd.dart';
+import 'package:test/test.dart';
+
+class LogicTestModule extends Module {
+  LogicTestModule(Logic a) {
+    addInput(a.name, a, width: a.width);
+  }
+}
+
+void main() {
+  test(
+      'GIVEN logic name is valid '
+      'THEN expected to see proper name being generated', () async {
+    final bus = Logic(name: 'validName');
+    final mod = LogicTestModule(bus);
+    await mod.build();
+    final sv = mod.generateSynth();
+    expect(sv, contains('input logic validName'));
+  });
+
+  test('GIVEN logic name is invalid THEN expected to see sanitized name',
+      () async {
+    final bus = Logic(name: '&*-FinvalidN11Me');
+    final mod = LogicTestModule(bus);
```

Comment:
You should be able to immediately check that `bus.name` has the sanitized value.  There are scenarios where users will be interested in the name of a `Logic` even before they generate SV.

Reply:
No need to generate the system verilog code

Reply:
Indeed a better approach, the test should be able to run much faster than need to generate system verilog code.